src/System.Runtime.Serialization.Xml/tests/DataContractSerializer.cs

```diff
@@ -1942,10 +1942,29 @@ public static void DCS_TypeWithCollectionWithoutDefaultConstructor()
     public static void DCS_DeserializeEmptyString()
     {
         var serializer = new DataContractSerializer(typeof(object));
-        Assert.Throws<XmlException>(() =>
+
+        bool exceptionThrown = false;
+        try
         {
             serializer.ReadObject(new MemoryStream());
-        });
+        }
+        catch (Exception e)
+        {
+            Type expectedExceptionType = typeof(XmlException);
+            Type actualExceptionType = e.GetType();
+            if (actualExceptionType != expectedExceptionType)
```

Comment:
Use .Equals as Type has an implementation of .Equals and it's what xunit is using. Also Type.Equals uses Type.UnderlyingSystemType which is abstract and does a ReferenceEquals against those type objects. If the types at runtime are different (e.g. RTType vs some other implementation), you might get different implementations of Type which return something different from UnderlyingSystemType. If it's in contract, try outputting the type information for the objects returned from UnderlyingSystemType and see if they are the same.